module/VuFind/src/VuFind/Controller/Plugin/AbstractRequestBase.php

```diff
@@ -303,6 +316,76 @@ public function getDefaultRequiredDate($checkHolds, $catalog = null,
         return $this->getDateFromArray($dateArray);
     }
 
+    /**
+     * Check if the user-provided dates are valid.
+     *
+     * Returns validated dates and/or an array of validation errors if there are
+     * problems.
+     *
+     * @param string $startDate       User-specified start date
+     * @param string $requiredBy      User-specified required-by date
+     * @param array  $extraHoldFields Hold form fields enabled by
+     * configuration/driver
+     *
+     * @return array
+     */
+    public function validateDates($startDate, $requiredBy, $extraHoldFields)
```

Comment:
Rather than having two parameters (`$startDate` and `$requiredBy`) here, what if you took an associative array (`['startDate' => ..., 'requiredBy' => ...]`), and then appended "TS" onto the keys for the output? This might be more flexible if different types of requests have different types of dates associated with them.

Reply:
You might also rename `$extraHoldFields` to something like `$enabledFormFields` to make this more generic for use with other types of requests in the future.

Reply:
Parameter renamed.